src/polyfillstub/intersection-observer-stub.js

```diff
@@ -35,11 +41,53 @@ const UPGRADERS = '_upgraders';
 export function shouldLoadPolyfill(win) {
   return (
     !win.IntersectionObserver ||
-    win.IntersectionObserver === IntersectionObserverStub ||
-    !win.IntersectionObserverEntry
+    !win.IntersectionObserverEntry ||
+    !!PolyfilledIntersectionObserver ||
+    !supportsDocumentRoot(win)
   );
 }
 
+/** @type {!IntersectionObserver} */
```

Comment:
I think the right type for this is `{typeof IntersectionObserver}` or `function(new: IntersectionObserver, ...)`